javaslang/src/test/java/javaslang/control/OptionTest.java

```diff
@@ -379,6 +380,29 @@ public void shouldNotConsumeAnythingOnPeekWhenValueIsNotDefined() {
         assertThat(testee).isEqualTo(Option.none());
     }
 
+    // -- transform
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowExceptionOnNullTransformFunction() {
+        Option<Integer> option = Option.some(1);
+        option.transform(null);
+    }
+
+    @Test
+    public void shouldApplyTransformFunctionToSome() {
+        Option<Integer> option = Option.some(1);
+        Function<Option<Integer>, String> f = o -> o.get().toString().concat("-transformed");
+        assertThat(option.<String>transform(f)).isEqualTo("1-transformed");
+    }
+
+    @Test
+    public void shouldHandleTransformOnNone() {
+        // TODO: What is the expected behavior when transform is called on None given return type may or may not be an Option?
+        // Calling None.get() will throw NoSuchElementException
+        // Should it be left to the caller to decide how to handle a value of None in their transform function?
+        // If not, should the default implementation of transform be overridden in None so that it is handled consistently?
```

Comment:
@kmarwood A good test for transform on None would be

```java
assertThat(Option.none().transform(self -> self.isEmpty() ? "ok" : "failed")).isEqualTo("ok");
```

Could you add that, please?